app/src/main/java/net/squanchy/search/view/ItemsAdapter.java

```diff
@@ -1,41 +1,135 @@
 package net.squanchy.search.view;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Locale;
 
-import net.squanchy.search.model.TitledList;
+import net.squanchy.search.SearchResults;
+import net.squanchy.search.view.SpeakerAdapter.ViewTypeId;
 import net.squanchy.speaker.domain.view.Speaker;
 
-import static net.squanchy.search.view.SpeakerAdapter.ViewTypeId;
-
 class ItemsAdapter {
 
-    private final List<TitledList<Speaker>> lists = new ArrayList<>();
+    // These values are "random", we count on them not clashing with the other
+    // random values that are used for non-hardcoded numeric IDs (for events
+    // and speakers). This is a reasonable assumption in the Long range.
+    private static final long ITEM_ID_EVENTS_HEADER = 100;
```

Comment:
another (safer) assumption is to return a negative number 👻 which is a valid item id and won't clash with hash functions (as far as I remember)